### PR TITLE
refactor: replace momentjs by date-fns

### PR DESCRIPTION
**Pull request by ocarreterom**
| Question         | Response    |
| ---------------- | ----------- |
| Version?         | v1.4.0      |
| Devices tested?  | iPhone 7 iOS 11.2 (simulator) |
| Bug fix?         | no      |
| New feature?     | no      |
| Includes tests?  | no      |
| All Tests pass?  | Running infinite      |
| Related ticket?  | #663        |

---

## Description
Use date-fns to represent relative time.

#### Some questions:
- Personally I like the long format (hours) more than the short one (h). Can be replaced easily.
- This PR have changes that affect translations. Only `en.js` has been updated (locales test not pass).
- I can't verify the tests have passed since it seems not work :/
captura de pantalla 2017-12-10 a las 22 31 06

### CODE REVIEW

**Comment by housseindjirdeh** [comment]
This is so great, thank you @ocarreterom <3

Yep the only thing I can think of is reverting back to single character representations of time as mentioned by @machour. Otherwise this looks beautiful.

**Comment by ocarreterom** [comment]
Comments addressed. Ready for review again :)

**Comment by machour** [comment]
@ocarreterom had a quick test with the last version of the PR, and it looks like you made it! 🎉 🎉 🎉 

Noticed a tiny glitch though, while testing your code on the simulator and the old one on my device, while looking at the events.

For events older than 1 hour, your PR was showing "1h" while the old code showed "2h".
Checked Github web, and it was also saying 2h. Refreshed several times, and right now, your PR started showing "2h".

Seems like timeago & date-fns don't make the same approximations when it comes to rounding.
date-fns, I think, considers "1h50" to be "1h" when momentjs reports it as "2h".

I don't think that this would be a show stopper, but I'm wondering if date-fns behavior can be fine-tuned with some configuration 🤔 

Also, could you take care of the rest of translations so that the Travis CI build passes? It should be pretty straight forward. Once done, I'll give this PR a wider test to validate it and it will be merged right after 1.4 is released.

Thank you again for your work, this one have been itching me for so long ❤️

**Comment by machour** [comment]
PS: You can reproduce that tiny shift by checking the time for the last comment here: https://github.com/machour/git-point-playground/issues/34#issuecomment-351651675
it's also the case for minutes.

Again, this is no biggie for me, but I'm not sure about the rest of the team.

**Comment by machour** [comment]
Reported this difference in this comparison thread: https://github.com/date-fns/date-fns/issues/275#issuecomment-351668021

**Comment by ocarreterom** [comment]
@machour thanks for your review.

Yes, you are right. The rounding is `floor` by default. I will change it to `round` and add the rest of translation just today. 👍 

I'm not sure it works exactly like moment.js since it uses custom range breakdowns 🙄 : https://momentjs.com/docs/#/displaying/fromnow/

**Comment by machour** [comment]
@ocarreterom Sorry I took so long to get back to you on this PR 😞 

I just gave it a new test and it's working as expected 🎉 🎉 🎉  

Last thing before being able to merge this PR would be to fix the Travis build. It seems that these two locale files needs to be updated:

- zh-tw.js
- eu.js

Could you take care of this please? 🙏 🙏 

Here are a few comparisons between this PR and GitHub web for the other reviewers to be amazed at:

screen shot 2018-01-15 at 10 25 01 pm

screen shot 2018-01-15 at 10 25 39 pm

**Comment by ocarreterom** [comment]
@machour no problem. It's ready :)

**Comment by machour** [comment]
Thank you for doing this so fast ❤️

**Comment by machour** [comment]
@housseindjirdeh ready for approval & merge!

File: package.json [hunk 1]
```diff
@@ -52,14 +52,14 @@
     ]
   },
   "dependencies": {
+    "date-fns": "^1.29.0",
```
Comment (housseindjirdeh) [inline package.json]: 🎉

**Comment by housseindjirdeh** [comment]
@machour Let's definitely explore possibilities to see if we can reduce bundle sizes if possible

**Pull request merged**